addon/components/sl-menu.js

```diff
@@ -232,415 +94,555 @@ export default Ember.Component.extend({
+                            break;
+
+                        case Direction.PARENT:
+                            this.selectParent();
+                            break;
+
+                        case Direction.PREVIOUS:
+                            this.selectPrevious();
+                            break;
+
+                        case Direction.RIGHT:
+                            this.selectRight();
+                            break;
+
+                        case Direction.SUB_MENU:
+                            this.selectSubMenu();
+                            break;
+
+                        case Direction.UP:
+                            this.selectUp();
+                            break;
+
+                        default:
+                            warn( `Received "select" stream action with invalid selection, "${indexOrDirection}"` );
+                            break;
+                    }
+                });
+            }
+        }
+    ),
 
     /**
-     * When true, allows key binding to drill down
+     * Retrieve the currently selected item
      *
-     * @type {Boolean}
+     * @function
+     * @returns {?Object}
      */
-    useDrillDownKey: true,
+    selectedItem: Ember.computed(
+        'selections.@each.item',
+        function() {
+            return this.get( 'selections.lastObject.item' );
+        }
+    ),
 
     // -------------------------------------------------------------------------
-    // Observers
+    // Methods
 
     /**
-     * Remove bound events and current menu state
+     * Clear the `selection` data
      *
      * @function
-     * @returns {undefined}
+     * @returns {Boolean} - True unless an error state is detected
      */
-    destroyMenu: Ember.on(
-        'willClearRender',
-        function() {
-            const parent = this.get( 'parentView' );
+    clearSelection() {
+        const selections = this.get( 'selections' );
 
-            if ( 'function' === Ember.typeOf( parent.unregisterChild ) ) {
-                parent.unregisterChild( this );
-            }
+        selections.forEach( ( selection ) => {
+            Ember.set( selection.item, 'selected', false );
+        });
 
-            const keyEvents = this.get( 'keyEvents' );
-            if ( keyEvents ) {
-                keyEvents.off( 'childSelected' )
-                    .off( 'drillDown' )
-                    .off( 'closeAll' )
-                    .off( 'showAll' )
-                    .off( 'cycleRootSelectionNext' )
-                    .off( 'cycleRootSelectionPrevious' );
-            }
-        }
-    ),
+        this.set( 'selection', null );
+
+        return true;
+    },
 
     /**
-     * Initialize children array
+     * Hide all the menu's sub-menus
      *
      * @function
      * @returns {undefined}
      */
-    initChildren: Ember.on(
-        'init',
-        function() {
-            this.set( 'children', new Ember.A() );
-        }
-    ),
+    hideAll() {
```

Comment:
Both hideAll() and showAll() don't really do any checking; they simply set an internal value. They are different from the select* methods since nothing is actually "selected". If you'd rather see Boolean returns, I can add them.